### ISSUE TITLE
A false positive about the rule UnsynchronizedStaticFormatter

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#4034]
**Affects PMD Version:6.46.0** 

**Rule:UnsynchronizedStaticFormatter**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_multithreading.html#unsynchronizedstaticformatter>

**Description:**
Hi, I found a false negative about the rule UnsynchronizedStaticFormatter. Please read the following code example. In line 2, there is an annotation `@Synchronized` and it can generate `synchronized` statement. Hence, I think this is a false negative.

**Code Sample demonstrating the issue:**
```java
private static final SimpleDateFormat sdf = new SimpleDateFormat();
@Synchronized("sdf")
void foo() {
    sdf.format(...);
}
```

**Expected outcome:**

PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*
